src/devices/Bmp180/Bmp180.cs

```diff
@@ -0,0 +1,261 @@
+        public Temperature ReadTemperature()
+        {
+            //Gets the compensated temperature in degrees celsius
+            short UT = ReadRawTemperature();
+
+            // Calculations below are taken straight from section 3.5 of the datasheet.
+            int X1 = (UT - _calibrationData.AC6) * _calibrationData.AC5 / (1 << 15);
+            int X2 = _calibrationData.MC * (1 << 11) / (X1 + _calibrationData.MD);
+            int B5 = (X1 + X2);
+            int T = (B5 + 8) / (1 << 4);
+            
+            return Temperature.FromCelsius((double)T/10);            
+        }
+
+        /// <summary>
+        ///  Reads the pressure from the sensor
+        /// </summary>
+        /// <returns>
+        ///  Atmospheric pressure in Pa
+        /// </returns>
+        public int ReadPressure()
+        {
+            //Gets the compensated pressure in Pascals
+            short UT = ReadRawTemperature();
+            int UP = ReadRawPressure();
+
+            // Calculations below are taken straight from section 3.5 of the datasheet.
+            // Calculate true temperature coefficient B5.
+            int X1 = (UT - (short)_calibrationData.AC6) * (short)_calibrationData.AC5 / (1 << 15);
+            int X2 = (short)_calibrationData.MC * (1 << 11) / (X1 + (short)_calibrationData.MD);
+            int B5 = (X1 + X2);
+
+            // Pressure Calculations
+            int B6 = B5 - 4000;
+            X1 = ((short)_calibrationData.B2 * (B6 * B6 / (1 << 12))) / (1 << 11);
+            X2 = (short)_calibrationData.AC2 * B6 / (1 << 11);
+            int X3 = X1 + X2;
+            int B3 = ((((short)_calibrationData.AC1 * 4 + X3) << (short)Sampling.Standard) + 2) / 4;
+            X1 = (short)_calibrationData.AC3 * B6 / (1 << 13);
+            X2 = ((short)_calibrationData.B1 * (B6 * B6 / (1 << 12))) / (1 << 16);
+            X3 = ((X1 + X2) + 2) / (1 << 2);
+            int B4 = _calibrationData.AC4 * (X3 + 32768) / (1 << 15);
+            int B7 = (UP - B3) * (50000 >> (short)Sampling.Standard);
+
+            int p = 0;
+            if((uint)B7 < 0x80000000)
+            {                
+                p = (int)(((long)B7 * 2) / B4);                
+            }
+            else
+            {               
+                p = (int)(((long)B7 / B4) * 2);             
+            }
+            
+            X1 = (p / (1 << 8)) * (p / (1 << 8));
+            X1 = (X1 * 3038) / (1 << 16);
+            X2 = (-7357 * p) / (1 << 16);
+            p = p + (X1 + X2 + 3791) / (1 << 4);
+
+            return p;
+        }
+
+        /// <summary>
+        ///  Calculates the altitude in meters from the specified sea-level pressure(in hPa).
+        /// </summary>
+        /// <param name="seaLevelPressure"> 
+        ///  Sea-level pressure in hPa
+        /// </param>
+        /// <returns>
+        ///  Height in meters from the sensor
+        /// </returns>
+        public double ReadAltitude(double seaLevelPressure)

```

Comment:
consider making this take a default value which uses average pressure (per wikipedia: 101.325 kPa)